pipelines/genericPipeline.py

```diff
@@ -0,0 +1,37 @@
+import threading
+from abc import ABC
+
+class GenericPipeline(ABC):
```

Comment:
Why not use [queue](https://docs.python.org/3/library/queue.html)?

Reply:
Threading is an accidental import here, don't need it

Reply:
Nothing to do with the threading library. What I mean is that the GenericPipeline class seems to re-implement the queue class.

Reply:
Class can be removed. From documentation:

> The Queue class in this module implements all the required locking semantics.